packages/react-query/src/useBaseQuery.ts

```diff
@@ -118,7 +118,9 @@ export function useBaseQuery<
   }
 
   // Handle result property usage tracking
-  return !defaultedOptions.notifyOnChangeProps
-    ? observer.trackResult(result)
-    : result
+  return React.useMemo(() => {
+    return !defaultedOptions.notifyOnChangeProps
+      ? observer.trackResult(result)
+      : result
+  }, [defaultedOptions.notifyOnChangeProps, observer, result])
```

Comment:
I don't believe this is an optimization ?
I mean, the initial code has a ternary expression that uses memoized result if the render wasn't abount a new value;
The target code has array and function creation every time + hook call that compares the previous and new array values.

Reply:
the goal of this PR is to return an object that is the same when the input is the same.